spec/app/controllers/course_excluded_exercises_controller/controller_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe CourseExcludedExercisesController, type: :request do
+  let(:request_payload) { 
+    {
+      course_uuid:        given_course_uuid,
+      sequence_number:    given_sequence_number,
+      exclusions:         given_exclusions,
+      request_uuid:       given_request_uuid
+    }
+  }
+
+  let(:given_exclusions) {
+    (given_specific_version_exclusions + given_any_version_exclusions).shuffle
+  }
+
+  let(:given_specific_version_exclusions) {
+    (number_of_exclusions - given_any_version_exclusions.length).times.map{
+      { 'exercise_uuid' => SecureRandom.uuid.to_s }
+    }
+  }
+
+  let(:given_any_version_exclusions) {
+    (Kernel::rand(number_of_exclusions)).times.map{
```

Comment:
Given that we do, ultimately, want to make sure we handle zero, one, and many exclusions, this sort of spec could be useful _at the service level_.  Or we could explicitly add specs for the zero, one, and many cases.  Or we could use the ruby equivalent of `hypothesis` (which I looked into, but it's not ready for prime-time yet).  Personally, I'd prefer to explicitly check the three cases directly rather than rely on a 1-in-N chance of catching a problem when running the specs.